src/jit/lclvars.cpp

```diff
@@ -235,6 +235,52 @@ void Compiler::lvaInitTypeRef()
 
     lvaInitArgs(&varDscInfo);
 
+#if FEATURE_FASTTAILCALL
+
+    //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned argRegCount      = 0;
+    unsigned floatingRegCount = 0;
+    unsigned stackArgCount    = 0;
+    size_t   stackSize        = 0;
+    unsigned compArgCount     = info.compArgsCount;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc) {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
+
+    unsigned   argNum;
+    LclVarDsc* curDsc;
+
+    for (curDsc = lvaTable, argNum = 0; argNum < varDscInfo.varNum; argNum++, curDsc++)
+    {
+        if (curDsc->lvIsRegArg)
+        {
+            incrementRegCount(curDsc);
+
+#if FEATURE_MULTIREG_ARGS
+            if (curDsc->lvOtherArgReg != REG_NA)
+            {
+                incrementRegCount(curDsc);
+            }
+#endif // FEATURE_MULTIREG_ARGS
+        }
+        else if (varTypeIsStruct(curDsc))
+        {
+            stackSize += curDsc->lvSize();
+        }
+        else
+        {
+            stackSize += TARGET_POINTER_SIZE;
+        }
+    }
```

Comment:
I have hoisted this into its own function lvArgStackSize. I agree with Carol that it is not ideal having this code in two different locations duplicated. I do not know of a better solution that does not involve refactoring fgMorphArgs. Any suggestions?